lib/ansible/modules/packaging/language/pip.py

```diff
@@ -234,6 +260,94 @@
 _SPECIAL_PACKAGE_CHECKERS = {'setuptools': 'import setuptools; print(setuptools.__version__)',
                              'pip': 'import pkg_resources; print(pkg_resources.get_distribution("pip").version)'}
 
+op_dict = {">=": operator.ge, "<=": operator.le, ">": operator.gt,
+           "<": operator.lt, "==": operator.eq, "!=": operator.ne}
+
+
+class Distribution:
+    """Python distribution package metadata wrapper.
+
+    A wrapper class for Requirement, which provides API to parse package name
+    and version specifier, providing fallback if the installed setuptools version
+    doesn't support required features.
+    """
+
+    def __init__(self, name_string, version_string=None):
+        self._plain_distribution = False
+        self._distribution_name = name_string
+        if name_string.startswith('file:') or _is_vcs_url(name_string):
+            return
+
+        self._plain_distribution = True
+        if version_string:
+            version_string = version_string.lstrip()
+            separator = '==' if version_string[0].isdigit() else ' '
+            name_string = separator.join((name_string, version_string))
+        self._requirement = Requirement.parse(name_string)
+        # old pkg_resource will replace 'setuptools' with 'distribute' when it already installed
+        if self._requirement.project_name == "distribute":
+            self._distribution_name = "setuptools"
+        else:
+            self._distribution_name = self._requirement.project_name
+
+    @property
+    def has_version_specifier(self):
+        if self._plain_distribution:
+            return bool(self._requirement.specs)
+        return False
+
+    @property
+    def distribution_name(self):
```

Comment:
This is unnecessary, you can write/access `self.distribution_name` directly and save one indirection.